### PR TITLE
Ft seller to delete product list #184651342

### PR DESCRIPTION
**Pull request by kagame8732**
#### What does this PR do?
A seller should be able to delete a specific item from their collection

#### Description of Task to be completed?
As a Seller
I should be able to delete a specific item in my collection
Such that I can remove items I no longer deal in/were erroneously added

#### How should this be manually tested?
-Open your favorite code editor
-Navigate to the project and clone it
-In the terminal type:
-`npm install` to install the required dependencies.
-`npm start` to start a server

by running 'npm run test' you can see the coverage

#### What are the relevant pivotal tracker/Trello stories?
[#184651342](https://www.pivotaltracker.com/n/projects/2633635)

### CODE REVIEW

File: src/controllers/deleteProduct.controller.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Product } from '../Database/models';
+
+const deleteProduct = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const userId = req.user.id;
+    const product = await Product.findOne({
+      where: { id, userId },
```
Comment (Raphaela664) [inline src/controllers/deleteProduct.controller.js:13]: - why do we have to ask for the userId?
- Consider the case where a merchant different from the one who created the product is trying to delete the product.

**Review by Raphaela664 — DISMISSED** [review]

File: src/controllers/deleteProduct.controller.js (hunk 1 above)
Comment (kagame8732) [reply src/controllers/deleteProduct.controller.js:13]: Because a merchant is supposed to delete their own product, they are not allowed to delete products belonging to others.

Comment (Raphaela664) [reply src/controllers/deleteProduct.controller.js:13]: There should be a condition to check if the logged in merchant is the one who wants to delete the product

**Review by Raphaela664 — DISMISSED** [review]
Great work! This PR looks good to me

File: .circleci/config.yml [hunk 2]
```diff
@@ -44,8 +44,7 @@ jobs:
           command: |
             ./cc-test-reporter format-coverage -t lcov ./coverage/lcov.info
             ./cc-test-reporter upload-coverage
-
 workflows:
   Test_and_Coverage:
     jobs:
-      - Test_and_Send_Coverage
+      - Test_and_Send_Coverage
```
Comment (didasmbalanya) [inline .circleci/config.yml]: Add a new empty line at the end of every file

File: src/controllers/deleteProduct.controller.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    await product.destroy();
+
+    res.status(200).json({
+      status: 'success',
+      message: 'Product deleted successfully',
+    });
+  } catch (err) {
+    console.log(err);
```
Comment (didasmbalanya) [inline src/controllers/deleteProduct.controller.js]: remove the console.log

File: src/controllers/deleteProduct.controller.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { Product } from '../Database/models';
+
+const deleteProduct = async (req, res) => {
+  try {
+    const { id } = req.params;
```
Comment (didasmbalanya) [inline src/controllers/deleteProduct.controller.js:5]: no validation. Will probably fail if id is not in the params

File: src/routes/deleteProduct.routes.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import express from 'express';
+import deleteProduct from '../controllers/deleteProduct.controller';
+import { authorizeUser } from '../middlewares/userRoles.middleware';
+import RoleCheck from '../middlewares/user.Checkrole';
+
+const router = express.Router();
+
+router.delete('/deleteProduct/:id', authorizeUser, RoleCheck(['merchant', 'admin']), deleteProduct);
```
Comment (didasmbalanya) [inline src/routes/deleteProduct.routes.js]: add a middleware validator for the product ID param

File: src/seeders/20230317073843-dummy-user.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable linebreak-style */
```
Comment (didasmbalanya) [inline src/seeders/20230317073843-dummy-user.js:1]: remove this

File: src/seeders/20230317073843-dummy-user.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @param {data} queryInterface .
+ * @returns {data} .
+ */
+export async function up(queryInterface) {
+  return queryInterface.bulkInsert('Dummy', [{
+    names: 'Dummy User',
+    email: 'dummy@email.com',
+    password: 'dummypassword',
```
Comment (didasmbalanya) [inline src/seeders/20230317073843-dummy-user.js:10]: this password needs to be hashed otherwise when logging in this dummy user the right way, it will fail

File: src/seeders/20230317073843-dummy-user.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ * @returns {data} .
+ */
+export async function down() {
+  /**
+   * Add commands to revert seed here.
+   *
+   * Example:
+   * await queryInterface.bulkDelete('People', null, {});
+   */
+}
```
Comment (didasmbalanya) [inline src/seeders/20230317073843-dummy-user.js:26]: undo the change you did on the UP function above. i.e buldDelete the user you seeded

File: src/tests/delete-product.test.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-unused-vars */
+/* eslint-disable linebreak-style */
```
Comment (didasmbalanya) [inline src/tests/delete-product.test.js]: remove these

**Review by didasmbalanya — DISMISSED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (kagame8732) [reply .circleci/config.yml]: done

File: src/controllers/deleteProduct.controller.js (hunk 3 above)
Comment (kagame8732) [reply src/controllers/deleteProduct.controller.js]: done

File: src/controllers/deleteProduct.controller.js (hunk 4 above)
Comment (kagame8732) [reply src/controllers/deleteProduct.controller.js:5]: done

File: src/routes/deleteProduct.routes.js (hunk 5 above)
Comment (kagame8732) [reply src/routes/deleteProduct.routes.js]: authorizeUser and RoleCheck are middleware

File: src/seeders/20230317073843-dummy-user.js (hunk 6 above)
Comment (kagame8732) [reply src/seeders/20230317073843-dummy-user.js:1]: done

File: src/seeders/20230317073843-dummy-user.js (hunk 7 above)
Comment (kagame8732) [reply src/seeders/20230317073843-dummy-user.js:10]: corrected

File: src/tests/delete-product.test.js (hunk 9 above)
Comment (kagame8732) [reply src/tests/delete-product.test.js]: done

**Review by bstwagir — APPROVED** [review]
Every review has been addressed.

**Review by JulesNIYOMWUNGERI — APPROVED** [review]
well!

**Review by Raphaela664 — APPROVED** [review]
